controller/replica_controller.go

```diff
@@ -270,6 +273,11 @@ func (rc *ReplicaController) syncReplica(key string) (err error) {
 			return err
 		}
 	}
+	// check whether the replica need to be scheduled
+	err = rc.scheduler.ScheduleReplica(replica)
```

Comment:
I think there is a problem with this approach.

Not sure if you observed it, but I think it may result in sometimes two replicas are scheduled on the same host while the other host has none. Since there are multiple workers working to update the replicas simultaneously, you may end up with multiple replicas was scheduled at the exact same time, and choosing the same node for the multiple replicas because when the schedule started, they cannot see the end result of others.

It's more reliable to do scheduling in the volume controller when we create the replica for the first time. It's in sequence for a single volume. https://github.com/rancher/longhorn-manager/blob/master/controller/volume_controller.go#L629